### PR TITLE
[Search] Rename HighWaterMark to the more generic ChangeTrackingState

### PR DESCRIPTION
**Pull request by Mohit-Chakraborty**

### CODE REVIEW

**Comment by Mohit-Chakraborty** [comment]
cc: @tg-msft

**Review by tg-msft — APPROVED** [review]
Looks good!

**Comment by gmndrg** [comment]
I believe this is clear enough, from my point of view. Thank you so much @Mohit ***@***.***>. Let's wait for dev feedback for final approval.

**Review by bleroy — APPROVED** [review]

**Pull request merged**
